### PR TITLE
Refactor save to reduce latency and overlap io with inferencing steps

### PR DESCRIPTION
**Pull request by daviddpruitt**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Comment by mnabian** [comment]
Hi @daviddpruitt , could you please push to the RC branch instead of main?

File: examples/generative/corrdiff/generate.py [hunk 1]
```diff
@@ -378,6 +379,17 @@ def generate_and_save(dataset, sampler, f: nc.Dataset, generate_fn, device, logg
     time_index = -1
     writer = writer_from_input_dataset(f, dataset)
 
+    # Initialize threadpool for writers
+    writer_executor = ThreadPoolExecutor(max_workers=2)
```
Comment (mnabian) [inline examples/generative/corrdiff/generate.py:383]: Why do you set the `num_workers` to `2`? Shouldn't this be configurable?

Comment (daviddpruitt) [reply examples/generative/corrdiff/generate.py]: Updating this to be a parameter in the configuration.

**Comment by mnabian** [comment]
@akshaysubr would you like to take another look and approve?

**Comment by akshaysubr** [comment]
@daviddpruitt Can you resolve the conflicts here? Some of them are in `generate.py` and it would be much easier to review after merging/rebasing with `main`.

**Review by akshaysubr — APPROVED** [review]

**Pull request closed without merging**